opennms-services/src/test/java/org/opennms/netmgt/poller/PollerRoutingTest.java

```diff
@@ -0,0 +1,94 @@
+package org.opennms.netmgt.poller;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.JndiRegistry;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
+import org.opennms.netmgt.poller.mock.MockMonitoredService;
+import org.opennms.netmgt.poller.monitors.AvailabilityMonitor;
+import org.springframework.test.context.ContextConfiguration;
+
+@RunWith( OpenNMSJUnit4ClassRunner.class )
+@ContextConfiguration( locations = { "classpath:/META-INF/opennms/emptyContext.xml" } )
+public class PollerRoutingTest extends CamelTestSupport {
+	
+	@Override
+    protected JndiRegistry createRegistry() throws Exception
+    {
+        JndiRegistry registry = super.createRegistry();
+
+        registry.bind( "availabilityMonitor", new AvailabilityMonitor() );
+        registry.bind( "availabilityMonitorCamel", new ServiceMonitorCamelImpl("direct:pollAvailabilityMonitor") );
+
```

Comment:
Don't override the createRegistry() method. Override the addServicesOnStartup() instead like this:
```
    @SuppressWarnings( "rawtypes" )
    @Override
    protected void addServicesOnStartup( Map<String, KeyValueHolder<Object, Dictionary>> services )
    {
        services.put( Pinger.class.getName(), new KeyValueHolder<Object, Dictionary>(new TestPinger(), new Properties()));
        ...
    }
```
The Properties argument holds the <service-properties> for the service.